File: cognition/src/main/scala/app/controllers/FortuneController.scala
package controllers

import scala.collection.mutable.ArrayBuffer

import org.cognition._
import conf.Authentication._
import models._
import views.fortunes._
import sqltyped._

class FortuneController extends Controller {

  get("fortunes") {
    withJDBC { _ => session =>
      implicit val conn = session.conn
      val buffer = new ArrayBuffer[Fortune]
      buffer += new Fortune(0, "Additional fortune added at request time.")
      sql("select id, message from fortune").apply
        .foreach(x => buffer += new Fortune(x.get("id"), x.get("message")))

      render.html(Index(buffer.sortBy(_.message)))
    }
  }

}

File: frameworks/PHP/kumbiaphp/bench/app/controllers/fortune_controller.php
<?php

class FortuneController extends AppController
{
    public function index()
    {
        View::select(null,'fortune');
        $data = Fortune::all('SELECT id, message FROM Fortune');
        $data[] = (object) ['id' => 0,'message' => 'Additional fortune added at request time.'];
        usort($data, function($a, $b){return strcmp($a->message, $b->message);});
        $this->data = $data;
    }
}
